src/main/java/it/pagopa/ecommerce/commons/domain/TransactionExpired.java

```diff
@@ -0,0 +1,45 @@
+package it.pagopa.ecommerce.commons.domain;
+
+import it.pagopa.ecommerce.commons.documents.TransactionExpiredEvent;
+import it.pagopa.ecommerce.commons.domain.pojos.BaseTransaction;
+import it.pagopa.ecommerce.commons.domain.pojos.BaseTransactionExpired;
+import it.pagopa.ecommerce.commons.domain.pojos.BaseTransactionWithClosureError;
+import it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto;
+import lombok.EqualsAndHashCode;
+
+/**
+ * <p>
+ * Expired transaction.
+ * </p>
+ *
+ * @see Transaction
+ * @see BaseTransactionWithClosureError
+ */
+@EqualsAndHashCode(callSuper = true)
+public final class TransactionExpired extends BaseTransactionExpired implements Transaction {
+
+    /**
+     * Primary constructor
+     *
+     * @param baseTransaction base transaction
+     * @param event           expiration event
+     */
+    public TransactionExpired(
+            BaseTransaction baseTransaction,
+            TransactionExpiredEvent event
+    ) {
+        super(baseTransaction, event.getData());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Transaction applyEvent(Object event) {
+        return this;
+    }
```

Comment:
Are we missing here the transition to TransactionRefunded? TransactionRefunded aggregate is not present as for now and maybe is better to add it into a separate PR, what do you think?

Reply:
Correct, and that's _exactly_ what I was thinking 😉